### ISSUE TITLE
[feat][OSS] Add clip_grad_value_ function.

### ISSUE THREAD

**Issue by dhkim0225** [facebookresearch/fairscale#308]
## 🚀 Feature

There are two popular gradient clipping methods: one that limits the maximum gradient value of each model parameter and the other one that scales the gradient value based on the p-norm of a (sub-)set of model parameters.
`clip_grad_norm`(the second one) is useful when the norm of gradients is large, but not when only a small sub-set of model parameters have abnormal gradient values since the norm will still be reasonably small considering the number of all model parameters.

Related PR (Pytorch Lightning): https://github.com/PyTorchLightning/pytorch-lightning/pull/5477

How about to add the following function to optim/oss.py

``` python
def clip_grad_value(
    self,
    clip_value: Union[float, int],
    filter_params_fn: Callable[[Any], Any] = None,
) -> None:
```

_I want to call function in PL's sharded_native_amp_plugin.py._

**Comment by blefaudeux** [comment]
Thanks for filing in this issue @dhkim0225, makes a lot of sense ! I can do that or feel free to file a PR, as you prefer. I'll be busy the next couple of days, but if this can wait then it can be on me

**Comment by dhkim0225** [comment]
@blefaudeux Sorry for late reply. I will be busy this week AGAIN.... 
@priancho has a lot of interest on fairscale. Would you mind If I ask you to make a PR for this?

**Comment by priancho** [comment]
@dhkim0225 sure. Let me see it :-)

**Comment by blefaudeux** [comment]
@dhkim0225 @priancho sorry for the time it took me to get back to you, been a bit busy.. After a second read, using the [PyTorch util](https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_value_.html?highlight=clip_grad_value_#torch.nn.utils.clip_grad_value_) on you model should just work, the sharding is not an issue here since the clipping is done on a per-parameter basis. I could add a util function as part of OSS but I think that it might be a bit confusing, it's not really needed in fact, unless I missed something ?

**Comment by dhkim0225** [comment]
@blefaudeux Thanks for your comment, Closing issue.
